packages/core/src/lib/markdown-it/highlight/helper.js

```diff
@@ -7,6 +7,52 @@ function splitCodeAndIndentation(codeStr) {
   return [indents, content];
 }
 
+const BOUNDARY_TYPE = {
+  Start: Symbol('start'),
+  End: Symbol('end'),
+};
+
+// Simplifies multiple bounds applied on a single line to an array of disjointed bounds
+// boundaryCollection:
+// e.g [{index: 1, type: BOUNDARY_TYPE.Start},
+//      {index:3, type: BOUNDARY_TYPE.End},
+//      {index: 5, type: BOUNDARY_TYPE.Start},
+//      {index: 7, type: BOUNDARY_TYPE.End}]
+function collateAllIntervals(boundaryCollection) {
```

Comment:
do we need to update documentation for this enhancement/bug fix? The only usage example there is obscenely long to be user friendly, so see if you can keep the edits minimal too.

https://markbind.org/userGuide/formattingContents.html#line-highlighting

If you'd like, feel free to split the example up into more comprehensible sections in this MR too.

Reply:
Overall great work and sleuthing, for a first issue especially.

Reply:
Alright I will see what I can cook up for the documentation. Thx.